src/main/kotlin/io/perfometer/dsl/HttpDsl.kt

```diff
@@ -20,78 +19,37 @@ import java.util.*
 sealed class HttpStep
 data class RequestStep(val request: HttpRequest) : HttpStep()
 data class PauseStep(val duration: Duration) : HttpStep()
+data class ParallelStep(val action: suspend () -> Unit) : HttpStep()
 
 typealias HttpHeader = Pair<String, String>
 typealias HttpParam = Pair<String, String>
 
-class RequestDsl(
-    private val url: URL,
-    private val method: HttpMethod,
-    initialHeaders: Map<String, String>
-) {
-    private var name: String? = null
-    private var path: String = ""
-    private val headers = initialHeaders.toMutableMap()
-    private val params = mutableListOf<HttpParam>()
-    private var body: ByteArray = ByteArray(0)
-    private var consumer: (HttpResponse) -> Unit = {}
-
-    fun name(name: String) {
-        this.name = name
-    }
-
-    fun path(path: String) {
-        this.path = path
-    }
-
-    fun body(body: ByteArray) {
-        this.body = body
-    }
-
-    fun headers(vararg headers: HttpHeader) {
-        this.headers.putAll(headers)
-    }
-
-    fun params(vararg params: HttpParam) {
-        this.params.addAll(params)
-    }
-
-    fun consume(consumer: (HttpResponse) -> Unit) {
-        this.consumer = consumer
-    }
-
-    private fun pathWithParams(): String {
-        return path + paramsToString()
-    }
-
-    private fun paramsToString(): String {
-        return if (this.params.isNotEmpty())
-            this.params.joinToString("&", "?") { "${it.first}=${it.second}" }
-        else ""
-    }
-
-    fun build() =
-        HttpRequest(name ?: "$method $path", method, url, pathWithParams(), headers, body, consumer)
-}
-
 class HttpDsl(
+    parallel: Boolean = false,
     private val baseURL: URL,
     private val scenarioRunner: ScenarioRunner,
 ) {
     private val headers = mutableMapOf<String, String>()
+    private val stepHandler: suspend (HttpStep) -> Unit = if (parallel) {
```

Comment:
It would be best to keep such logic in ScenarioRunner. Lets try do it like this:
* Define `typealias HttpVisitor = (HttpStep) -> Unit` and pass it to `HttpDsl` instead of `ScenarioRunner`
* Pass different visitors depending on usage of HttpDsl (when running parallel step in `ScenarioRunner` you would construct a visitor running steps asynchronously)

It would be also good to have limited version of HttpDsl available in `parallel` block (running `pause` or another `parallel` in `paralles` does not make any sense) but we could leave that to fix later.